bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/protocol/SoulissBindingSendDispatcherJob.java

```diff
@@ -0,0 +1,465 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class provide to take packet, and send it to regular interval to Souliss
+ * Network
+ *
+ * @author Tonino Fazio
+ * @since 1.7.0
+ */
+@NonNullByDefault
+public class SoulissBindingSendDispatcherJob implements Runnable {
+
+    private final Logger logger = LoggerFactory.getLogger(SoulissBindingSendDispatcherJob.class);
+
+    @Nullable
+    private SoulissGatewayHandler gw;
+    static boolean bPopSuspend = false;
+    protected static ArrayList<SoulissBindingSocketAndPacketStruct> packetsList = new ArrayList<>();
+    private long startTime = System.currentTimeMillis();
+    static String ipAddressOnLAN = "";
+    static int iDelay = 0; // equal to 0 if array is empty
+    static int sendMinDelay = 0;
+
+    public SoulissBindingSendDispatcherJob(Bridge bridge) {
+        gw = (SoulissGatewayHandler) bridge.getHandler();
+        ipAddressOnLAN = gw.ipAddressOnLAN;
+    }
+
+    /**
+     * Put packet to send in ArrayList PacketList
+     */
+    public static synchronized void put(@Nullable DatagramSocket socket, DatagramPacket packetToPUT, Logger logger) {
+        bPopSuspend = true;
+        boolean bPacchettoGestito = false;
+        // estraggo il nodo indirizzato dal pacchetto in ingresso
+        // restituisce -1 se il pacchetto non è del tipo SOULISS_UDP_FUNCTION_FORCE
+        int node = getNode(packetToPUT);
+        if (node >= 0) {
+            logger.debug("Push packet in queue - Node {}", node);
+        }
+        /*
+         * else {
+         * // logger.debug("Push packet in queue");
+         * }
+         */
+
+        if (packetsList.size() == 0 || node < 0) {
+            bPacchettoGestito = false;
+        } else {
+            // OTTIMIZZATORE
+            // scansione lista pacchetti da inviare
+            for (int i = 0; i < packetsList.size(); i++) {
+                if (node >= 0 && getNode(packetsList.get(i).packet) == node && !packetsList.get(i).isSent()) {
+                    // frame per lo stesso nodo già  presente in lista
+                    logger.debug("Frame UPD per nodo {} già presente in coda. Esecuzione ottimizzazione.", node);
+                    bPacchettoGestito = true;
+                    // se il pacchetto da inserire è più corto (o uguale) di
+                    // quello in coda allora sovrascrivo i byte del pacchetto
+                    // presente nella coda
+                    if (packetToPUT.getData().length <= packetsList.get(i).packet.getData().length) {
+                        // scorre i byte di comando e se il byte è diverso da
+                        // zero sovrascrive il byte presente nel pacchetto in
+                        // coda
+                        logger.debug("Optimizer. Packet to push: {}", macacoToString(packetToPUT.getData()));
+                        logger.debug("Optimizer. Previous frame: {}",
```

Comment:
If these are multiple lines of output they should be TRACE and not DEBUG.

Reply:
fixed